python-package/setup.py

```diff
@@ -101,6 +105,12 @@ def compile_cpp(use_mingw=False, use_gpu=False, use_mpi=False,
     logger.info("Starting to compile the library.")
 
     cmake_cmd = ["cmake", "../compile/"]
+    if opencl_python_package:
+        use_gpu = False
+        use_mpi = False
+        nomp = False
+        use_hdfs = False
```

Comment:
Again, I don't think we need this as this build option is for maintainers.

Reply:
I remove everything again except the use_gpu which I think is a useful guard.